.github/main.yml

```diff
@@ -0,0 +1,47 @@
+name: Actions
+on: [push]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Begin CI...
+        uses: actions/checkout@v2
+
+      - name: Use Node 12
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+
+      - name: Use cached node_modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: nodeModules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            nodeModules-
```

Comment:
you need to cache all node_modules to work properly

like this (circleci example)

```jsx
paths:
            - ~/.cache/yarn
            - ./node_modules
            - ./packages/server/node_modules
....

    
          - ./packages/web/node_modules
```